packages/@dcl/sdk/src/with-composite.ts

```diff
@@ -1,19 +1,30 @@
-import { engine, Composite } from '@dcl/ecs'
-import { createContentFetchCompositeProvider } from './composite-provider'
-import { onStart as basicOnStart, onUpdate as basicOnUpdate } from './index'
+import { Composite, engine } from '@dcl/ecs'
+import { crdtGetState } from '~system/EngineApi'
+import { compositeProvider } from './composite-provider'
+import { onUpdate, rendererTransport } from './index'
 
-export const onUpdate = basicOnUpdate
+export { onUpdate }
 
 /**
  * @internal
  * Function that is called before the first update and after the evaluation of the code.
  */
 /* @__PURE__ */ export async function onStart() {
-  const compositeProvider = await createContentFetchCompositeProvider()
-  const mainComposite = compositeProvider.getCompositeOrNull('main.composite')
-  if (mainComposite) {
-    Composite.instance(engine, mainComposite, compositeProvider)
+  const response = await crdtGetState({ data: new Uint8Array() })
+
+  // when this condition is true something like `main.crdt` was pre-loaded from the runtime, we don't need to instance the main.composite
+  if (!response.hasEntities) {
+    const mainComposite = compositeProvider.getCompositeOrNull('main.composite')
```

Comment:
should we make this the default and _only_ behavior? It will be easier to reason if "if there is a main.composite it will always be loaded". that way we won't need the --customEntryPoint

Reply:
make sense to me, but `--customEntryPoint` is for those that want to expose their own `onUpdate` and `onStart`